### PR TITLE
feat: add authorization list to rpc transaction and tx receipt types

### PR DESCRIPTION
**Pull request by PanGan21**
## Motivation

Adds 7702 authorization list to rpc transaction and transaction receipt types.
Related: #1032 

## PR Checklist

- [X] Added Tests
- [x] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

File: crates/rpc-types-eth/src/transaction/mod.rs [hunk 1]
```diff
@@ -103,7 +106,10 @@ pub struct Transaction {
     )]
     #[doc(alias = "tx_type")]
     pub transaction_type: Option<u8>,
-
+    /// The authorization list is a list of tuples that store the address to code which the signer
+    /// desires to execute in the context of their EOA.
+    #[serde(skip_serializing_if = "Option::is_none")]
```
Comment (mattsse) [inline crates/rpc-types-eth/src/transaction/mod.rs:111]:
```suggestion
    #[serde(default, skip_serializing_if = "Option::is_none")]
```

File: crates/rpc-types-eth/src/transaction/receipt.rs [hunk 2]
```diff
@@ -56,6 +57,10 @@ pub struct TransactionReceipt<T = ReceiptEnvelope<Log>> {
     /// EIP98 makes this optional field, if it's missing then skip serializing it
     #[serde(skip_serializing_if = "Option::is_none", rename = "root")]
     pub state_root: Option<B256>,
+    /// The authorization list is a list of tuples that store the address to code which the signer
+    /// desires to execute in the context of their EOA.
+    #[serde(skip_serializing_if = "Option::is_none")]
```
Comment (mattsse) [inline crates/rpc-types-eth/src/transaction/receipt.rs:62]:
```suggestion
    #[serde(default, skip_serializing_if = "Option::is_none")]
```

File: crates/rpc-types-eth/src/transaction/mod.rs [hunk 3]
```diff
@@ -103,7 +106,10 @@ pub struct Transaction {
...
+    pub authorization_list: Option<Vec<Authorization>>,
```
Comment (mattsse) [inline crates/rpc-types-eth/src/transaction/mod.rs:112]: I believe we actually want the SignedAuthoriazion here because we also need the signature

**Review by mattsse — CHANGES_REQUESTED** [review]
ty!

**Review by DaniPopes — APPROVED** [review]

File: crates/rpc-types-eth/src/transaction/mod.rs (hunk 3 above)
Comment (PanGan21) [reply crates/rpc-types-eth/src/transaction/mod.rs]: Updated

Comment (PanGan21) [reply crates/rpc-types-eth/src/transaction/mod.rs]: Seems fixed!

**Pull request merged**